src/test/regress/expected/multi_modifying_xacts.out

```diff
@@ -119,8 +119,7 @@ EXCEPTION
         RAISE NOTICE 'caught not_null_violation';
 END $$;
 NOTICE:  caught not_null_violation
-COMMIT;
-ERROR:  cannot ROLLBACK TO SAVEPOINT in transactions which modify distributed tables
+ROLLBACK;
 \set VERBOSITY default
```

Comment:
I think this means that the savepoint rollback logic needs to reset ->transactionFailed too (when it's been "triggered" inside the relevant savepoint(s)).